app/scripts/utils/functional.es6

```diff
@@ -36,6 +36,33 @@ function asArray(obj) {
     .map(k => [k, obj[k]]);
 }
 
+const checkData = (data) => {
+  return data ? Ok(data) : Err('There is not data');
+};
+
+/**
+ * this function shall be use to aggregate into add the current item base on aggProp
+ * @param {object} agg The initial accumulator value
+ * @param {object} item The data to process
+ * @param {string} aggProp the property name to aggregate
+ * @param {string} countProp the property name to count
+ * @returns {any} The accumulated value
+ */
+function sumAggregate(agg, item, aggProp, countProp) {
+  const aggPropVal = item[aggProp];
+  const sumPropVal = item[countProp];
+  if (!agg[aggPropVal]) {
```

Comment:
watch out for implicit boolean casts -- javascript has really weird edge cases. `lodash/lang/isUndefined` handles these well.